### PR TITLE
Fix the caching is not supported on this batch system error (resolves #931)

### LINKED ISSUE #931: Add more helpful error message if caching unsupported batch system is used
Stems from #928

### PR DESCRIPTION
**Pull request by arkal**
resolves #931

### CODE REVIEW

**Comment by arkal** [comment]
@hannes-ucsc

File: src/toil/common.py [hunk 1]
```diff
@@ -661,7 +661,9 @@ def createBatchSystem(config, jobStore=None, userScript=None):
             raise RuntimeError('Unrecognised batch system: %s' % config.batchSystem)
 
         if not (config.disableSharedCache or batchSystemClass.supportsWorkerCleanup()):
-            raise RuntimeError('%s currently does not support shared caching.' % config.batchSystem)
+            raise RuntimeError('%s currently does not support shared caching.  Use the '
+                               '--disableSharedCache flag when running Toil to use this batch '
```
Comment (hannes-ucsc) [inline src/toil/common.py]:
```
Use the --disableSharedCache option when running Toil with this batch system.
```

**Comment by hannes-ucsc** [comment]
Commit message is too long.

File: src/toil/common.py [hunk 2]
```diff
@@ -661,7 +661,9 @@ def createBatchSystem(config, jobStore=None, userScript=None):
             raise RuntimeError('Unrecognised batch system: %s' % config.batchSystem)
 
         if not (config.disableSharedCache or batchSystemClass.supportsWorkerCleanup()):
-            raise RuntimeError('%s currently does not support shared caching.' % config.batchSystem)
+            raise RuntimeError('%s currently does not support shared caching.  Use the '
+                               '--disableSharedCache flag when running Toil with this batch '
```
Comment (hannes-ucsc) [inline src/toil/common.py]: `option`, not `flag`

**Comment by hannes-ucsc** [comment]
@arkal, please remove `needs work` label when you think this is ready to review.

**Comment by arkal** [comment]
@hannes-ucsc it's ready

**Comment by hannes-ucsc** [comment]
Jenkins, test this please.

**Comment by hannes-ucsc** [comment]
Jenkins, test this please.

**Comment by hannes-ucsc** [comment]
The build triggered by https://github.com/BD2KGenomics/toil/pull/932#issuecomment-224394775 hung. See #953 for follow-up.

**Comment by hannes-ucsc** [comment]
Build failure tracked in #906.

Commit message is still too long. Look at where GitHub wraps it on this page. We can't have that. Sorry.

**Comment by hannes-ucsc** [comment]
@arkal

**Comment by arkal** [comment]
Hahaha i thought you were joking (sarcastically) about the lack of a "message" in the commit since the "title" is self explanatory. I'll fix this now

**Comment by arkal** [comment]
@hannes-ucsc i implemented the suggested changes

**Pull request merged**